src/server/mod.rs

```diff
@@ -683,14 +685,14 @@ impl Stream for AddrIncoming {
                     return Ok(Async::Ready(Some(AddrStream::new(socket, addr))));
                 },
                 Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => return Ok(Async::NotReady),
-                Err(ref e) if self.sleep_on_errors => {
+                Err(ref e) if self.sleep_on_errors.is_some() => {
```

Comment:
I wonder if this can be rewritten as:

```rust
Err(ref e) if let Some(delay) = self.sleep_on_errors => {
```

so you don't need to use `.unwrap()` later.

Reply:
Tried that, but the compiler does not allow `let` in a match arm, as far as I can see.

Reply:
Could be done like this, right?

```rust
match self.listener.accept() {
    // Ok ...
    Err(e) => {
        if let Some(delay) = self.sleep_on_errors {
            // continue or delay
        }
        return Err(e);
    }
}
```